main_test.go

```diff
@@ -34,7 +34,7 @@ func (suite *PokemonTestSuite) TestGetPokemonByID() {
 	suite.router.ServeHTTP(w, req)
 
 	assert.Equal(suite.T(), http.StatusOK, w.Code)
-	assert.Equal(suite.T(), "{\"id\":1,\"name\":\"bulbasaur\"}", w.Body.String())
+	assert.Equal(suite.T(), "{\"id\":1,\"name\":\"bulbasaur\",\"image\":\"\"}", w.Body.String())
 }
 
 func (suite *PokemonTestSuite) TestGetPokemonByInvalidID() {
```

Comment:
Consider using Table Testing Drive, as it should create more complex scenarios and easiness on creating the testcases, also this is not a unit test as you are creating the full repository, this will be hard to test and will cause you that all the test cases would require the test data file

Reply:
Indeed, these tests seem to align more with integration testing rather than unit testing. I have relocated the tests to the Pokemon repository and adopted the Table Testing Drive approach.